### PR TITLE
[FLINK-32852][JUnit5 Migration] Migrate the scheduler package of flink-runtime module to JUnit5

### PR DESCRIPTION
**Pull request by RocMarshal**
## What is the purpose of the change

[FLINK-32852][JUnit5 Migration] Migrate the scheduler package of flink-runtime module to JUnit5

## Brief change log

[FLINK-32852][JUnit5 Migration] Migrate the scheduler package of flink-runtime module to JUnit5

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by RocMarshal** [comment]
hi, @JingGe Could you help take a look if you had the free time ?
Thank you very much~

File: flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InternalMiniClusterExtension.java [hunk 1]
```diff
@@ -109,4 +117,24 @@ public Object resolveParameter(
...
+
+    @Override
+    public void before(ExtensionContext context) throws Exception {
+        miniClusterResource.before();
+    }
+
+    @Override
+    public void after(ExtensionContext context) throws Exception {
+        miniClusterResource.after();
+    }
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InternalMiniClusterExtension.java:139]: I saw the mini cluster resource didn't close for each test before. I guess one reason is performance.

Start a mini cluster needs some times, if we restart it frequently, test will take a long time. 

Is it necessary for this PR?

image

File: flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraphTest.java [hunk 2]
```diff
@@ -120,9 +123,8 @@ public void testOnGloballyTerminalStateCalled() throws Exception {
...
         context.close();
 
-        assertThat(
-                stateWithExecutionGraph.getGloballyTerminalStateFuture().get(),
-                is(JobStatus.FINISHED));
+        assertThatFuture(stateWithExecutionGraph.getGloballyTerminalStateFuture())
+                .isCompletedWithValue(JobStatus.FINISHED);
     }
 
     @Test
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraphTest.java:130]: All public can be removed.

File: flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/BackgroundTaskTest.java [hunk 3]
```diff
@@ -18,55 +18,52 @@
...
     @Test
     public void testFinishedBackgroundTaskDoesNotContainAResult() {
         final BackgroundTask<Void> finishedBackgroundTask = BackgroundTask.finishedBackgroundTask();
 
-        assertTrue(finishedBackgroundTask.getResultFuture().isCompletedExceptionally());
+        assertThatFuture(finishedBackgroundTask.getResultFuture()).isCompletedExceptionally();
     }
 
     @Test
     public void testNormalCompletionOfBackgroundTask() {
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/BackgroundTaskTest.java:60]: All public can be removed.

File: flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SsgNetworkMemoryCalculationUtilsTest.java [hunk 4]
```diff
@@ -182,21 +182,20 @@ private void triggerComputeNumOfSubpartitions(IntermediateResult result) {
     private void assertNetworkMemory(
             List<SlotSharingGroup> slotSharingGroups, List<MemorySize> networkMemory) {
 
-        assertEquals(slotSharingGroups.size(), networkMemory.size());
+        assertThat(networkMemory.size()).isEqualTo(slotSharingGroups.size());
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SsgNetworkMemoryCalculationUtilsTest.java:185]: hasSameSizeAs

File: flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SsgNetworkMemoryCalculationUtilsTest.java [hunk 5]
```diff
@@ -270,16 +269,16 @@ private void createExecutionGraphAndEnrichNetworkMemory(
...
 
     private static JobGraph createJobGraph(
             final List<SlotSharingGroup> slotSharingGroups,
             List<Integer> parallelisms,
             ResultPartitionType resultPartitionType) {
 
-        assertThat(slotSharingGroups.size(), is(3));
-        assertThat(parallelisms.size(), is(3));
+        assertThat(slotSharingGroups.size()).isEqualTo(3);
+        assertThat(parallelisms.size()).isEqualTo(3);
```
Comment (1996fanrui) [inline flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SsgNetworkMemoryCalculationUtilsTest.java:281]: hasSize

**Review by 1996fanrui — COMMENTED** [review]
Some classes should be changed, but they are missed, so I commented here:

- Remove @ExtendWith(TestLoggerExtension.class)
  - EnforceMinimalIncreaseRescalingControllerTest
  - DefaultExecutionDeployerTest
  - StopWithSavepointTest
- Don't use `org.junit.Assert`
  - StateValidator
  - SchedulerTestingUtils
  - MockStateWithExecutionGraphContext
- public can be removed for test method
  - AdaptiveSchedulerTest
  - StateLocalitySlotAssignerTest
-  public can be removed for test class
  - you can search `public class`, I saw a lot of classes need to remove it.
- Update `isEqualTo(0)` to `isZero`
  - AdaptiveSchedulerTest
  - AllToAllBlockingResultInfoTest
 - Update `isEqualTo(1)` to `isOne`
   - ExecutionTimeBasedSlowTaskDetectorTest
   - SpeculativeExecutionTest
   - PointwiseBlockingResultInfoTest
   - DefaultVertexParallelismAndInputInfosDeciderTest
   - AdaptiveSchedulerTest
   - AllToAllBlockingResultInfoTest
 - hasSameSizeAs
   - DefaultVertexParallelismAndInputInfosDeciderTest (line 445)

**Comment by RocMarshal** [comment]
> public can be removed for test method
AdaptiveSchedulerTest
StateLocalitySlotAssignerTest

Thank you very much for the patient-check and review.

There're some class referenced by other class. so I kept the `public` key-word for the related class and removed the key-word for some class.

As shown in the commit messages, I split the patches by commit.
I'd be appreciated with your next confirmation!

**Comment by 1996fanrui** [comment]
@flinkbot run azure

**Review by 1996fanrui — APPROVED** [review]
Thanks @RocMarshal for the update!

LGTM

**Pull request merged**